r/R/install-arrow.R

```diff
@@ -62,55 +62,50 @@ install_arrow <- function(nightly = FALSE,
-      }
+  if (conda) {
+    if (nightly) {
+      system("conda install -y -c arrow-nightlies -c conda-forge --strict-channel-priority r-arrow")
     } else {
-      Sys.setenv(
-        LIBARROW_DOWNLOAD = "true",
-        LIBARROW_BINARY = binary,
-        LIBARROW_MINIMAL = minimal,
-        ARROW_R_DEV = verbose,
-        ARROW_USE_PKG_CONFIG = use_system
-      )
-      # On the M1, we can't use the usual autobrew, which pulls Intel dependencies
-      apple_m1 <- grepl("arm-apple|aarch64.*darwin", R.Version()$platform)
-      # On Rosetta, we have to build without JEMALLOC, so we also can't autobrew
-      rosetta <- identical(sysname, "darwin") && identical(system("sysctl -n sysctl.proc_translated", intern = TRUE), "1")
-      if (rosetta) {
-        Sys.setenv(ARROW_JEMALLOC = "OFF")
-      }
-      if (apple_m1 || rosetta) {
-        Sys.setenv(FORCE_BUNDLED_BUILD = "true")
-      }
+      system("conda install -y -c conda-forge --strict-channel-priority r-arrow")
+    }
+  } else {
+    Sys.setenv(
+      LIBARROW_DOWNLOAD = "true",
+      LIBARROW_BINARY = binary,
+      LIBARROW_MINIMAL = minimal,
+      ARROW_R_DEV = verbose,
+      ARROW_USE_PKG_CONFIG = use_system
+    )
+    # On the M1, we can't use the usual autobrew, which pulls Intel dependencies
+    apple_m1 <- grepl("arm-apple|aarch64.*darwin", R.Version()$platform)
+    # On Rosetta, we have to build without JEMALLOC, so we also can't autobrew
+    rosetta <- identical(sysname, "darwin") && identical(system("sysctl -n sysctl.proc_translated", intern = TRUE), "1")
+    if (rosetta) {
+      Sys.setenv(ARROW_JEMALLOC = "OFF")
+    }
+    if (apple_m1 || rosetta) {
+      Sys.setenv(FORCE_BUNDLED_BUILD = "true")
+    }
 
-      opts <- list()
-      if (apple_m1 || rosetta) {
-        # Skip binaries (esp. for rosetta)
-        opts$pkgType <- "source"
-      } else if (isTRUE(binary)) {
-        # Unless otherwise directed, don't consider newer source packages when
-        # options(pkgType) == "both" (default on win/mac)
-        opts$install.packages.check.source <- "no"
-        opts$install.packages.compile.from.source <- "never"
-      }
-      if (length(opts)) {
-        old <- options(opts)
-        on.exit(options(old))
-      }
-      install.packages("arrow", repos = arrow_repos(repos, nightly), ...)
+    opts <- list()
+    if (apple_m1 || rosetta) {
+      # Skip binaries (esp. for rosetta)
+      opts$pkgType <- "source"
+    } else if (isTRUE(binary)) {
+      # Unless otherwise directed, don't consider newer source packages when
+      # options(pkgType) == "both" (default on win/mac)
+      opts$install.packages.check.source <- "no"
+      opts$install.packages.compile.from.source <- "never"
     }
-    if ("arrow" %in% loadedNamespaces()) {
-      # If you've just sourced this file, "arrow" won't be (re)loaded
-      reload_arrow()
+    if (length(opts)) {
+      old <- options(opts)
+      on.exit(options(old))
```

Comment:
This isn't enough to justify it, and I don't think that it's a recursive dependency of anything we currently depend on, but this makes me long for `withr::with_options()`

Reply:
Sure 🤷 

FTR all I did here is remove the `if` indentation, no substantive change to this code

Reply:
I don't think we should just for this + right now, but reading it and making sure what was intended took a bit. Maybe next time we reach for a similar pattern

Reply:
We Suggest packages that require withr but don't Import any. We could, in the meantime, add a `with_options()` to util.R, though FWIW this is the only place we'd be using it right now in the package.